src/rules/extensions.ts

```diff
@@ -370,10 +370,18 @@ export default createRule<Options, MessageId>({
             })
             const fixOrSuggest = {
               fix(fixer: TSESLint.RuleFixer) {
-                return fixer.replaceText(
-                  source,
-                  replaceImportPath(source.raw, fixedImportPath),
+                const newImportPath = replaceImportPath(
```

Comment:
The fix should be checking `fix: source.raw === newImportPath ? null : () => {}`